### PR TITLE
fix provAPIServer when NodePort svc is stuck waiting for lb endpoint

### PR DESCRIPTION
**Pull request by rewantsoni**
When the ProviderAPIServer is of type NodePort, the ocs operator waits for LoadBalancer endpoint to be available, which should not be the case. add a condition so that when service is of type nodePort, this check is not done.
The PR also updates the StoragProviderEndpoint appropriately based on the type of the Service.

### CODE REVIEW

**Review by iamniting — CHANGES_REQUESTED** [review]
Can you pls create a method for finding the endpoint for the load balancer type of service? It will reduce nested if conditions. 
Also can you pls have a commit-msg not more than 73 char?

File: controllers/storagecluster/provider_server.go [hunk 1]
```diff
@@ -185,27 +186,79 @@ func (o *ocsProviderServer) createService(r *StorageClusterReconciler, instance
...
+			return reconcile.Result{}, err
+		}
+
+		if len(nodeAddresses) == 0 {
+			err = fmt.Errorf("Did not found any worker node addresses")
+			r.Log.Error(err, "Worker nodes count is zero")
+			return reconcile.Result{}, err
+		}
+
+		instance.Status.StorageProviderEndpoint = nodeAddresses[0] + ":" + fmt.Sprint(ocsProviderServiceNodePort)
```
Comment (iamniting) [inline controllers/storagecluster/provider_server.go:211]: Can you pls use the same format as line no. 198

File: controllers/storagecluster/provider_server.go [hunk 2]
```diff
@@ -185,27 +186,79 @@ func (o *ocsProviderServer) createService(r *StorageClusterReconciler, instance
...
+		instance.Status.StorageProviderEndpoint = fmt.Sprintf("%s:%d", endpoint, ocsProviderServicePort)
+	} else {
+		nodeAddresses, err := o.getWorkerNodesInternalIPAddresses(r)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if len(nodeAddresses) == 0 {
+			err = fmt.Errorf("Did not found any worker node addresses")
+			r.Log.Error(err, "Worker nodes count is zero")
```
Comment (iamniting) [inline controllers/storagecluster/provider_server.go:207]: This does not seem an error to me, Can you pls rephrase it?

File: controllers/storagecluster/provider_server.go [hunk 3]
```diff
@@ -185,27 +186,79 @@ func (o *ocsProviderServer) createService(r *StorageClusterReconciler, instance
...
+		instance.Status.StorageProviderEndpoint = nodeAddresses[0] + ":" + fmt.Sprint(ocsProviderServiceNodePort)
+
+	}
+
+	r.Log.Info("status.storageProviderEndpoint is updated", "Endpoint", instance.Status.StorageProviderEndpoint)
+
+	return reconcile.Result{}, nil
+}
+
+func (o *ocsProviderServer) getLoadBalancerEndpoint(service *corev1.Service) string {
```
Comment (iamniting) [inline controllers/storagecluster/provider_server.go:220]:
```suggestion
func (o *ocsProviderServer) getLoadBalancerServiceEndpoint(service *corev1.Service) string {
```

**Review by iamniting — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/provider_server.go (hunk 1 above)
Comment (rewantsoni) [reply controllers/storagecluster/provider_server.go]: done

File: controllers/storagecluster/provider_server.go (hunk 3 above)
Comment (rewantsoni) [reply controllers/storagecluster/provider_server.go]: done

File: controllers/storagecluster/provider_server.go (hunk 2 above)
Comment (rewantsoni) [reply controllers/storagecluster/provider_server.go:207]: done

File: controllers/storagecluster/provider_server.go [hunk 4]
```diff
@@ -185,27 +186,79 @@ func (o *ocsProviderServer) createService(r *StorageClusterReconciler, instance
...
+
+		instance.Status.StorageProviderEndpoint = fmt.Sprintf("%s:%d", endpoint, ocsProviderServicePort)
+	} else {
+		nodeAddresses, err := o.getWorkerNodesInternalIPAddresses(r)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if len(nodeAddresses) == 0 {
+			err = fmt.Errorf("No workers can be found")
```
Comment (iamniting) [inline controllers/storagecluster/provider_server.go:206]:
```suggestion
			err = fmt.Errorf("Did not found any worker nodes")
```

**Review by iamniting — CHANGES_REQUESTED** [review]

File: controllers/storagecluster/provider_server.go (hunk 4 above)
Comment (rewantsoni) [reply controllers/storagecluster/provider_server.go]: done

**Review by iamniting — APPROVED** [review]

**Pull request merged**